examples/cpp/navigation/global_navigation/include/global_navigation.hpp

```diff
@@ -0,0 +1,96 @@
+/****************************************************************************
+ * Copyright (c) 2023 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+#pragma once
+
+
+#include <rclcpp/rclcpp.hpp>
+#include <px4_ros2/navigation/experimental/global_navigation_interface.hpp>
+#include <px4_ros2/common/context.hpp>
+
+#include <Eigen/Core>
+
+using namespace std::chrono_literals; // NOLINT
+
+class GlobalNavigationTest : public px4_ros2::Context
+{
+public:
+  explicit GlobalNavigationTest(rclcpp::Node & node)
+  : Context(node), _node(node)
+  {
+    // Instantiate global navigation interface
+    _global_navigation_interface = std::make_shared<px4_ros2::GlobalNavigationInterface>(*this);
```

Comment:
I agree that for the user just passing a `Node` to the interface to get things running is ideal. But In that case if we're not passing a `Context` as a constructor argument to the interface, how will the interface get hold of a Context? I believe one of the future uses of the Context for the interface had to do with registration with the FMU. But maybe this can be done directly in the interface, though I think that would be inconsistent with the modes implementation.

@bkueng maybe you have some insight on this? Does it make sense to have the interface just take a `Node` for initialisation?

Reply:
Or perhaps the interface can inherit from `Context` in that case? Then the Context is abstracted away, but still available internally

Reply:
Yes the idea is to do something like this:
Library:
```
class NavigationInferaceBase : public Context {
...

class GlobalNavigationInferaceBase : public NavigationInferaceBase {
...
```
User:
```
class GlobalNavigationTest : public px4_ros2::GlobalNavigationInferaceBase {
...
```